QWeb/keywords/text.py

```diff
@@ -862,23 +862,28 @@ def get_text(
     \`Is Text\`, \`VerifyText\`
     """
     anchor_int = util.anchor_to_index(anchor)
-    tag = kwargs.get('tag', None)
+    tag = kwargs.pop('tag', None)
     if tag:
         try:
             kwargs['element_kw'] = True
             shadow_dom = CONFIG['ShadowDOM']
             if shadow_dom:
-                web_element = internal_text.get_items_including_shadow_dom(locator, tag)[anchor_int]
+                web_element = internal_text.get_items_including_shadow_dom(locator,
+                                                                           tag,
+                                                                           **kwargs)[anchor_int]
             else:
                 web_element = element.get_elements_by_attributes(tag, locator, **kwargs)[anchor_int]
         except ValueError as e:
             raise QWebValueError(
                 'Only index is allowed anchor when searching element by it\'s attribute') from e
     elif '//' not in locator:
-        web_element = internal_text.get_text_using_anchor(locator, anchor)
+        # css=False means we include non-clickable texts
+        web_element = internal_text.get_text_using_anchor(locator, anchor, css=False, **kwargs)
     else:
         web_element = element.get_unique_element_by_xpath(locator, index=anchor_int, **kwargs)
-    text = web_element.text
+    # selenium's .text does not see all texts directly under slots
+    # use innerText as a backup
+    text = web_element.text or web_element.get_attribute('innerText')
```

Comment:
Probably should add `or ""` ternary because text could become `None` if `web_element.text == ""` and there's no attribute `innerText`
If the text becomes `None` then the `get_substring()` at the end of function will make `None` to show up as elements text

```
text = "" or None
text is None
```
->True

```
text = "" or None or ""
text is None
```
-> False

`web_element.text or web_element.get_attribute('innerText') or ""`

Reply:
It seems updates to this PR are no longer working (were done while GitHub had problems).  I will close this one and open another one